tests/components/screenlogic/test_init.py

```diff
@@ -272,22 +273,26 @@ def add_ents(ents: list[Any]) -> None:
         is_connected=True,
         _async_connected_request=DEFAULT,
     ):
-        connect_info = await async_get_connect_info(hass, mock_config_entry)
-        gateway = ScreenLogicGateway()
-        await gateway.async_connect(**connect_info)
-
-        mock_coordinator = Mock(
-            spec=ScreenlogicDataUpdateCoordinator,
-            config_entry=mock_config_entry,
-            gateway=gateway,
-            hass=hass,
+        assert await hass.config_entries.async_setup(mock_config_entry.entry_id)
+        await hass.async_block_till_done()
+
+        coordinator: ScreenlogicDataUpdateCoordinator = hass.data[DOMAIN][
+            mock_config_entry.entry_id
+        ]
+        gateway: ScreenLogicGateway = coordinator.gateway
+
+        assert (
+            hass.states.get(f"{BINARY_SENSOR_DOMAIN}.{device_prefix}_active_alert")
+            is not None
+        )
+        assert (
+            hass.states.get(f"{SENSOR_DOMAIN}.{device_prefix}_air_temperature")
+            is not None
         )
-        hass.data.setdefault(DOMAIN, {})[mock_config_entry.entry_id] = mock_coordinator
-        for platform in PLATFORMS:
-            component = cast(
-                ComponentProtocol,
-                importlib.import_module(
-                    f"homeassistant.components.screenlogic.{platform}"
-                ),
+        if EQUIPMENT_FLAG.CHLORINATOR in gateway.equipment_flags:
+            assert (
```

Comment:
Don't use any conditions in tests. We should know how the gateway is configured since we set up the test.

Reply:
I was using parameterize to test against different gateway configurations, but as this test is really supposed to only be testing setup with missing data, I can build a single data set to support that.

Reply:
We can extract the entity_ids that we want to assert state for, into another parameter. Then do a for loop here over those entity_ids.

Reply:
6d7eae3